### ISSUE TITLE
App says it doesn't have the permission to use location

### ISSUE THREAD

**Issue by kacperek585** [tijmenvangulik/Ergometerspace#24]
I was trying to connect to PM5 via bluetooth but the erg was claiming that connection is successful and my phone was pairing forever... I thought its because of the firmware so I updated the PM5. Now both PM5 and my phone claim to be connected but the app is unable to find an ergometer 😂 I gave it all the permissions possible

**Comment by tijmenvangulik** [comment]
I have heart about these problems before. It seems to be related to the hardware limitations. (if you have also other blue tooth devices connected, it may fix it if you disconnect them) On my phone I do not have these problems so this makes it very hard for me to fix it. I know about another developer who had the same problem and he fixed it but never published the fix. I ask him if he wants to publish his fix.

Tijmen

**Comment by kacperek585** [comment]
Thanks!

**Comment by tijmenvangulik** [comment]
I am now experiencing the same problem (for the first time) Hopefully I can now find a fix for this problem. 

Tijmen

**Comment by tijmenvangulik** [comment]
As (temporary) work around for now you could use a usb connection or use the web app from a pc or a mac.

Tijmen

**Comment by tijmenvangulik** [comment]
I found a work around for phones with limited blue tooth hardware. I am now working on a fix.

**Comment by tijmenvangulik** [comment]
I have now published an update. I found out that there  permission problem on androd 10 and above. I have implemented this fix. But since I do not own a android 10 device, I can not test it. Can you please check if this fix is working for you?

Tijmen

**Comment by kacperek585** [comment]
I have an Android 9 but I will try 😂

**Comment by tijmenvangulik** [comment]
If it does not work, I have also an extra feature added in the options: "Work around limited blue tooth hardware” . Switch it on and restart your app. Make sure that you have given the app all the rights.

TIjm

**Comment by tijmenvangulik** [comment]
I am now a step further. I have upgraded my android device to android 10. (and nearly destroyed my phone, but after a weekend getting all sorts of work arounds I got it working  ) And yes I have the same problem as you have on android 10. My app can not connect using blue tooth or usb. When I open ergometer-space.org in the google chrome browser , blue tooth works fine. So there is something with the app which needs upgrading...
